components/BottomAppBar/src/private/MDCBottomAppBarView.m

```diff
@@ -0,0 +1,254 @@
+      if (self.layoutDirection == UIUserInterfaceLayoutDirectionLeftToRight) {
+        floatingButtonPoint = CGPointMake(width - kMDCBottomAppBarFloatingButtonPositionX,
+                                          halfDefaultDimension);
+      } else {
+        floatingButtonPoint = CGPointMake(kMDCBottomAppBarFloatingButtonPositionX,
+                                          halfDefaultDimension);
+      }
+      break;
+    }
+    default:
+      break;
+  }
+  return floatingButtonPoint;
+}
+
+- (void)cutBottomAppBarViewAnimated:(BOOL)animated {
+  CGPathRef cutPath =
+      [self.bottomBarLayer pathWithCutFromRect:self.bounds
+                        floatingButtonPosition:self.floatingButtonPosition
+                               layoutDirection:self.layoutDirection];
+  if (animated) {
+    CABasicAnimation *pathAnimation =
+        [CABasicAnimation animationWithKeyPath:kMDCBottomAppBarViewPathString];
+    pathAnimation.duration = kMDCFloatingButtonExitDuration;
+    pathAnimation.fromValue = (id)self.bottomBarLayer.presentationLayer.path;
+    pathAnimation.toValue = (__bridge id _Nullable)(cutPath);
+    pathAnimation.fillMode = kCAFillModeForwards;
+    pathAnimation.removedOnCompletion = NO;
+    pathAnimation.delegate = self;
+    [pathAnimation setValue:kMDCBottomAppBarViewPathString
+                     forKey:kMDCBottomAppBarViewAnimKeyString];
+    [self.bottomBarLayer addAnimation:pathAnimation forKey:kMDCBottomAppBarViewPathString];
+  } else {
+    self.bottomBarLayer.path = cutPath;
+  }
+}
+
+- (void)healBottomAppBarViewAnimated:(BOOL)animated  {
+  CGPathRef withoutCutPath =
+      [self.bottomBarLayer pathWithoutCutFromRect:self.bounds
+                           floatingButtonPosition:self.floatingButtonPosition
+                                  layoutDirection:self.layoutDirection];
+  if (animated) {
+    CABasicAnimation *pathAnimation =
+        [CABasicAnimation animationWithKeyPath:kMDCBottomAppBarViewPathString];
+    pathAnimation.duration = kMDCFloatingButtonEnterDuration;
+    pathAnimation.fromValue = (id)self.bottomBarLayer.presentationLayer.path;
+    pathAnimation.toValue = (__bridge id _Nullable)(withoutCutPath);
+    pathAnimation.fillMode = kCAFillModeForwards;
+    pathAnimation.removedOnCompletion = NO;
+    pathAnimation.delegate = self;
+    [pathAnimation setValue:kMDCBottomAppBarViewPathString
+                     forKey:kMDCBottomAppBarViewAnimKeyString];
+    [self.bottomBarLayer addAnimation:pathAnimation forKey:kMDCBottomAppBarViewPathString];
+  } else {
+    self.bottomBarLayer.path = withoutCutPath;
+  }
+}
+
+- (void)animationDidStop:(CAAnimation *)animation finished:(BOOL)flag {
+  if (flag) {
+    [self renderPathBasedOnFloatingButtonVisibitlityAnimated:NO];
+    NSString *animKeyString = [animation valueForKey:kMDCBottomAppBarViewAnimKeyString];
+    if ([animKeyString isEqualToString:kMDCBottomAppBarViewPathString]) {
+      [self.bottomBarLayer removeAllAnimations];
```

Comment:
Do you really want to remove all animations or just the ones you're positive you started?

Reply:
Removing specific animations.